io_scene_niftools/modules/nif_export/animation/transform.py

```diff
@@ -94,6 +94,29 @@ def export_kf_root(self, b_armature=None):
             #         # wipe controller target
             #         ctrl.target = None
 
+        # skyrim
+        elif bpy.context.scene.niftools_scene.game in ('SKYRIM'):
+
+            NifLog.info(f"SKYRIM: Exporting animation on the skeleton: {b_armature.name}")
+            b_action = self.get_active_action(b_armature)
+            kf_root = block_store.create_block("NiControllerSequence")
+            targetname = "Scene Root"
+            for bone in b_armature.data.bones:
+                self.export_transforms(kf_root,b_armature,b_action,bone)
+            anim_textextra = self.export_text_keys(b_action)
+
+            kf_root.name = b_action.name
```

Comment:
No it does not need `if b_armature:` check. In nif_export.py this function is called and the armature is provided to this function when the animation only option is selected in the export panel.
`if NifOp.props.animation in ('ANIM_KF', 'ALL_NIF_XNIF_XKF'):
                NifLog.info("Creating keyframe tree")
                kf_root = self.transform_anim.export_kf_root(b_armature)`

Reply:
It will return a error only when there is no armature in the blender scene and that is the correct response, kf file can't be exported without the armature

Reply:
Should we throw a specific exception at this point then to inform the user.

Reply:
Ok I have edited the code.